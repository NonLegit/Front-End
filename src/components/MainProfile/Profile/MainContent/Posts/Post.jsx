import { Box } from '@mui/material';
import ArticleOutlinedIcon from '@mui/icons-material/ArticleOutlined';
import { useState, useContext, useEffect } from 'react';

import { useNavigate } from 'react-router-dom';
import { useEditPostContext } from '../../../../../contexts/EditPostContext';
import { CommunitiesContext } from '../../../../../contexts/CommunitiesModeratorContext';
import {
  EmptyImage,
  Flair,
  PostContentBox,
  PostImage,
  PostSidebaRes,
  PostsQueueBox,
  TagPost,
  TitlePost,
} from './styles';
import PostSide from './PostSide/PostSide';
import PostHeader from './PostHeader/PostHeader';
import PostFooter from './PostFooter/PostFooter';
import { UserContext } from '../../../../../contexts/UserProvider';
import { CommunitiesSubscriberContext } from '../../../../../contexts/CommunitiesSubscriberContext';
import { actionOnPost, moderationAction } from '../../../profileServer';

/**
 * the post that appear in posts - saved - hidden - upvoted - downvotep taps
 *
 * @component Post
 * @property {object} entity - an entity conatins all post informations
 * @property {string} type -subTitle of the page
 * @returns {React.Component} Post
 */
function Post(props) {
  const {
    entity,
    type,
  } = props;
  const [expand, setExpand] = useState();
  const [subTitle, setSubTitle] = useState(type);
  const [modList, setModList] = useState(false);
  const { username } = useContext(UserContext);
  const { communities } = useContext(CommunitiesContext);
  const { communitiesSubscriber } = useContext(CommunitiesSubscriberContext);
  const [notJoined, setNotJoined] = useState(false);

  const [isNsfw, setIsNsfw] = useState(entity?.nsfw);
  const [isSpoiler, setIsSpoiler] = useState(entity?.spoiler);
  const [isLocked, setIsLocked] = useState(entity?.locked);
  const [modState, setModState] = useState(entity?.modState);

  const handleExpand = () => {
    setExpand((prev) => !prev);
  };
  useEffect(() => {
    console.log(entity?.postVoteStatus);
    setSubTitle(type);
    if (communitiesSubscriber?.filter((element) => element.fixedName === entity.owner.name).length === 0) {
      setNotJoined(true);
    }
    if (entity.ownerType === 'Subreddit' && communities?.filter((element) => element.fixedName === entity.owner.name).length > 0) {
      setModList(true);
    }
  }, [type, communities, entity, communitiesSubscriber]);

  const handleNsfw = () => {
    actionOnPost(entity?.postid, isNsfw ? 'unmark_nsfw' : 'mark_nsfw');
    setIsNsfw((prev) => !prev);
  };
  const handleSpoiler = () => {
    actionOnPost(entity?.postid, isSpoiler ? 'unspoiler' : 'spoiler');
    setIsSpoiler((prev) => !prev);
  };
  const handleLock = () => {
    actionOnPost(entity?.postid, isLocked ? 'unlock_comments' : 'lock_comments');
    setIsLocked((prev) => !prev);
  };
  const handleApprove = () => {
    moderationAction(entity?.postid, 'approve');
    setModState('approved');
  };
  const handleRemove = () => {
    moderationAction(entity?.postid, 'remove');
    setModState('removed');
  };
  const handleSpam = () => {
    moderationAction(entity?.postid, 'spam');
    setModState('spammed');
  };

  const navigate = useNavigate();
  const { setEditPost } = useEditPostContext();

  return (
    <PostsQueueBox saved={subTitle}>
      <PostSide postid={entity?._id} points={entity?.votes} postVoteStatus={entity?.postVoteStatus} spam={entity?.modState === 'spam'} />

      <PostSidebaRes>
        <Box sx={{ display: 'flex' }}>
          <EmptyImage onClick={() => { setEditPost(false); navigate(`/${entity?.ownerType === 'Subreddit' ? 'subreddit' : 'user'}/${entity?.owner?.name}/comments/${entity?._id}`); }}>
            {entity?.images.length === 0 ? (

              <ArticleOutlinedIcon fontSize="small" color="disabled" />
            )
              : (
                <PostImage src={entity?.images[0]} alt="post pic" />

              )}
          </EmptyImage>
          <PostContentBox>
            <Box sx={{ marginLeft: 1 }}>
              <Box sx={{ display: 'flex', alignItems: 'center', flexWrap: 'wrap' }} onClick={() => { setEditPost(false); navigate(`/${entity?.ownerType === 'Subreddit' ? 'subreddit' : 'user'}/${entity?.owner?.name}/comments/${entity?._id}`); }}>
                <TitlePost variant="h6">{entity?.title}</TitlePost>
                {
                  entity?.flairId?.text
                  && (
                  <Flair
                    disableRipple
                    backgroundcolor={entity?.flairId?.backgroundColor}
                    flaircolor={entity?.flairId?.textColor}
                  >
                    {entity?.flairId?.text}
                  </Flair>
                  )
                }
                {isSpoiler && <TagPost color="#A4A7A8" variant="caption">spoiler</TagPost>}
                {isNsfw && <TagPost color="#FF585B" variant="caption">nsfw</TagPost>}
              </Box>
              <PostHeader
                subReddit={entity?.owner.name}
                type={entity?.ownerType}
                nameUser={entity?.author.name}
                Time={entity?.createdAt}
                locked={isLocked}
                modState={modState}
                notJoined={notJoined}
                sharedFrom={entity?.sharedFrom}
              />
              <PostFooter
                postid={entity?._id}
                ownerType={entity?.ownerType}
                owner={entity?.owner?.name}
                handleExpand={handleExpand}
                expand={expand}
                submitted={subTitle === undefined}
                isModList={modList}
                postedByOthers={(entity?.author.name !== username)}
                saved={entity?.isSaved}
                hidden={entity?.isHidden}
                numComments={entity?.commentCount}
                points={entity?.votes}
                postVoteStatus={entity?.postVoteStatus}
                sendReplies={entity?.sendReplies}
                nsfw={isNsfw}
                spoiler={isSpoiler}
                locked={isLocked}
                modState={modState}
                handleLock={handleLock}
                handleSpoiler={handleSpoiler}
                handleNsfw={handleNsfw}
                handleApprove={handleApprove}
                handleRemove={handleRemove}
                handleSpam={handleSpam}
              />
            </Box>
          </PostContentBox>
        </Box>
        {expand && <Box dangerouslySetInnerHTML={{ __html: entity?.text }} />}
      </PostSidebaRes>
    </PostsQueueBox>
  );
}

export default Post;
