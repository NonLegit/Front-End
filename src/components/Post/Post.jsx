/* eslint-disable react/no-danger */
// mui components
import {
  Box, useMediaQuery, useTheme,
} from '@mui/material';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';

// styles
import { useRef, useEffect, useState } from 'react';

import { useNavigate } from 'react-router-dom';
import {
  PostContainer, PostMedia, CustomImage, PostText, PostTextContainer, ControlsIcon, PostUrl, PostUrlLink, LinkIcon,

} from './styles';

import Reactions from './Reactions/Reactions';
import PostReactions from './PostReactions/PostReactions';
import PostHeader from './PostHeader/PostHeader';
import SharedPost from './SharedPost/SharedPost';
/**
 * This component is the view of the post in home page.
 *
 * @component Post
 * @property {string} title -Post title.
 * @property {string} ownerType -Post owner type user or subreddit.
 * @property {string} ownerIcon -Post owner icon.
 * @property {string} ownerName -Post subreddit(post owner).
 * @property {string} authorName -Post author.
 * @property {string} flairText -Post flair text.
 * @property {string} flairBackgroundColor -Post flair background color.
 * @property {string} flairColor -Post flair color.
 * @property {string} kind -Post kind (link, self, image, video).
 * @property {Array.<string>} images -Array of sources of images
 * @property {Array.<string>} videos -Array of sources of vidoes
 * @property {number} votes -Number of post votes.
 * @property {number} commentCount -Number of post comments.
 * @property {string} text -Post text in case of "self" kind.
 * @property {boolean} subreddit -to identify if post in home page or subreddit.
 * @property {number} postVoteStatus -The last vote of the current user in this post.
 * @property {boolean} isSaved -Is this post saved by the current user or not.
 * @property {number} postId -The Id of the current post.
 * @property {string} url -The post url.
 * @property {boolean} nsfw -Whether the post is not safe for work or not.
 * @property {boolean} spoiler -Whether the post is spoiler or not.
 * @property {Post} sharedFrom -the Parent post of the current one.
 * @returns {React.Component} Post
 */

function Post(props) {
  const {
    createdAt, title, images, ownerType, ownerName, ownerIcon, authorName, flairText, flairBackgroundColor, flairColor, kind, votes, commentCount, text, video,
    subredit, postVoteStatus, isSaved, postId, url, nsfw, spoiler, sharedFrom,
  } = props;

  // routes
  const navigate = useNavigate();

  // styles
  const theme = useTheme();
  const matchSm = useMediaQuery(theme.breakpoints.up('sm'));
  const matchMd = useMediaQuery(theme.breakpoints.up('md'));

  // variables
  const maxTextHeight = 180;

  // states
  const [displayShadow, setDisplayShadow] = useState(false);
  const [index, setIndex] = useState(0);
  const [maxImagesHeight, setMaxImagesHeight] = useState(450);

  // refs
  const postTextRef = useRef();
  const postMediaRef = useRef();

  // handlers
  const handleDirection = (dir) => {
    setIndex(index + dir);
  };

  const redirectToPost = (redirect, shared) => {
    if (redirect) {
      const username = ownerName;
      if (ownerType === 'User') {
        if (username) {
          navigate(`/user/${username}/comments/${shared ? sharedFrom?._id : postId}`);
        } else {
          navigate('/');
        }
      } else {
        navigate(`/subreddit/${ownerName}/comments/${shared ? sharedFrom?._id : postId}`);
      }
    }
  };

  // effects
  useEffect(() => {
    setDisplayShadow(postTextRef?.current?.offsetHeight > maxTextHeight);
  }, [text]);

  useEffect(() => {
    images?.forEach((image) => {
      const img = new Image();

      img.src = image;
      img.onload = () => {
        setMaxImagesHeight((maxImagesHeight) => {
          const maxValue = Math.min(maxImagesHeight, img.height);
          const postWidth = postMediaRef?.current?.offsetWidth;
          if (maxImagesHeight > img.height && img.width > postWidth) {
            return img.height * (postWidth / img.width);
          }
          return maxValue;
        });
      };
    });
  }, [images]);
  if (sharedFrom) {
    console.log('posty shared', props);
  }

  // console.log('for post ', postId, maxImagesHeight);
  return (
    <PostContainer my={2}>
      {matchSm && (
        <Reactions
          flexDirection="column"
          votes={votes}
          postVoteStatus={postVoteStatus}
          postId={postId}
        />
      )}
      <Box
        p={1}
        flexGrow={1}
        maxWidth={matchSm ? '94.5%' : '100%'}
      >
        <PostHeader
          title={title}
          ownerIcon={ownerIcon}
          ownerName={ownerName}
          authorName={authorName}
          flairText={flairText}
          flairBackgroundColor={flairBackgroundColor}
          flairColor={flairColor}
          createdAt={createdAt}
          subredit={subredit}
          ownerType={ownerType}
          nsfw={nsfw}
          spoiler={spoiler}
          redirectToPost={redirectToPost}
          sharedFrom={!!sharedFrom}
        />
        {/* eslint-disable jsx-a11y/media-has-caption */}
        {/* */}
        {!nsfw && (
        <PostMedia
          mt={1.5}
          kind={kind}
          ref={postMediaRef}
          spoiler={spoiler}
          onClick={() => redirectToPost(kind !== 'link', sharedFrom)}
        >
          {!sharedFrom ? (kind === 'video' ? (
            <video controls style={{ width: '100%', maxHeight: '512px' }}>
              <source src={video} type="video/mp4" />
            </video>
          ) : (
            (kind === 'image')
              ? (
                <>
                  {images.map((image, imageIndex) => (
                    imageIndex === index
                    && (
                    <CustomImage
                      src={image}
                      alt="post image"
                      key={image}
                      maxHeight={maxImagesHeight}
                    />
                    )
                  ))}
                  <>
                    <ControlsIcon
                      disableRipple
                      left={10}
                      display={index <= 0 ? 'none' : 'flex'}
                      sx={{
                        boxShadow: 10,
                      }}
                      color="third"
                      onClick={(e) => {
                        e.stopPropagation();
                        handleDirection(-1);
                      }}
                    >
                      <ArrowBackIosNewIcon />
                    </ControlsIcon>
                    <ControlsIcon
                      disableRipple
                      right={10}
                      display={index >= images.length - 1 ? 'none' : 'flex'}
                      sx={{
                        boxShadow: 10,
                      }}
                      color="third"
                      onClick={(e) => {
                        e.stopPropagation();
                        handleDirection(1);
                      }}
                    >
                      <ArrowForwardIosIcon />
                    </ControlsIcon>
                  </>
                </>
              ) : ((kind === 'self') ? (
                <PostText
                  ref={postTextRef}
                  maxHeight={displayShadow ? maxTextHeight : 'none'}
                >
                  {displayShadow && <PostTextContainer />}
                  <div dangerouslySetInnerHTML={{ __html: text }} />
                </PostText>
              )
                : (
                  <PostUrl href={url}>
                    <PostUrlLink>
                      {url}
                    </PostUrlLink>
                    <LinkIcon />
                  </PostUrl>
                ))
          ))
            : (
              <SharedPost
                sharedFrom={sharedFrom}
                subredit={subredit}
              />
            )}
        </PostMedia>
        )}
        <PostReactions
          votes={votes}
          matchSm={matchSm}
          matchMd={matchMd}
          comments={commentCount}
          postVoteStatus={postVoteStatus}
          isSaved={isSaved}
          postId={sharedFrom?._id || postId}
          redirectToPost={redirectToPost}
          authorName={authorName}
          subredit={subredit}
        />
      </Box>
    </PostContainer>
  );
}

export default Post;
