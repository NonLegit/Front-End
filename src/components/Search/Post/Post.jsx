import {
  Avatar, Box, Typography,
} from '@mui/material';
import { useState } from 'react';
import numFormatter from '../../../utils/MembersNum';
import calculateTime from '../../../utils/calculateTime';
import {
  PostContainer, PostInfo, PostInfoLink, PostTitle,
  PostRich, PostMedia, Image, Video, Flair, PostUrl, PostUrlLink, LinkIcon,
  // Flair,
} from './styles';

/**
 * Posts in search by posts
 * @component

 * @return {React.Component} - Posts in search by posts
 */

function Post(props) {
  const { post } = props;
  const [type] = useState('');

  return (
    <PostContainer my={2} sx={{ '&:hover': { border: 1 }, padding: '8px 16px 4px' }}>
      <Box width="100%">
        <PostInfo py={1}>
          <Avatar
            src={post?.owner?.icon}
            sx={{
              width: 20,
              height: 20,
            }}
            alt="Profile Image"
          />
          <PostInfoLink to={`/r/${post?.owner?.fixedName}`} color="#000" fontWeight="bolder">
            r/
            {post.owner.name}
          </PostInfoLink>
          <Box color="#787C7E" fontWeight={300} display="flex" gap="4px" flexWrap="wrap">
            <span>
              â€¢
            </span>
            <div>Posted By</div>
            <Avatar
              src={post?.author?.profilePicture}
              sx={{
                width: 20,
                height: 20,
              }}
              alt="Profile Image"
            />
            <PostInfoLink to={`/user/${post?.author?.userName}`} color="inherit" fontWeight="normal">
              u/
              {post?.author?.displayName}
              {' '}
              {calculateTime(post?.createdAt)}
            </PostInfoLink>
          </Box>
        </PostInfo>

        <PostTitle to={`../subreddit/${post?.owner?.fixedName}/comments/${post?._id}`}>
          <Typography variant="div" component="div" sx={{ fontSize: 18, wordBreak: 'break-all' }}>
            {post.title.replace(/<[^>]+>/g, '')}
            {/* {post?.title} */}

            {
            post?.flairId
            && (
            <Flair
              disableRipple
              backgroundColor={post?.flairId?.backgroundColor}
              flairColor={post?.flairId?.textColor}
            >
              {/* {post.flairId.text.replace(/<[^>]+>/g, '')} */}
              {post?.flairId?.text}
            </Flair>
            )
          }
          </Typography>
          {(post.kind === 'video' || type === 'image')
          && (
          <PostMedia>
            {(post.kind === 'video')
            && <Video src={post?.video} alt="" />}
            {(type === 'image')
             && <Image src={post?.images[0]} alt="" />}
          </PostMedia>
          )}

        </PostTitle>
        {(post.kind === 'link')
            && (
            <PostUrl href={post?.url}>
              <PostUrlLink>
                {post?.url}
              </PostUrlLink>
              <LinkIcon />
            </PostUrl>
            )}
        <PostRich mt={5}>
          <Typography variant="span" sx={{ marginRight: 3 }}>
            {numFormatter(post?.votes)}
            {' '}
            upvotes
            {' '}
          </Typography>
          <Typography variant="span" sx={{ marginRight: 3 }}>
            {numFormatter(post?.commentCount)}
            {' '}
            comments
            {' '}
          </Typography>
        </PostRich>
      </Box>

    </PostContainer>
  );
}

export default Post;
