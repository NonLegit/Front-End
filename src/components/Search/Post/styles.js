import { Box } from '@mui/material';
import { styled } from '@mui/material/styles';
import { Link } from 'react-router-dom';
import LaunchIcon from '@mui/icons-material/Launch';
import RedditButton from '../RedditButton/RedditButton';

export const PostContainer = styled(Box)(() => ({
  width: '100%',
  backgroundColor: '#fff',
  display: 'flex',
  borderRadius: 3,
  border: '1px solid #ccc',
  alignItems: 'center',
  '&:hover':
  {
    borderColor: '#999',
  },
  margin: 0,
}));

export const PostInfo = styled(Box)(() => ({
  display: 'flex',
  fontSize: '12px',
  gap: '4px',
}));

export const PostInfoLink = styled(Link)(({ color, fontWeight }) => ({
  fontWeight,
  color,
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
}));

export const PostTitle = styled(Link)(() => ({
  textDecoration: 'none',
  display: 'flex',
  flexDirection: 'row',
  alignItems: 'spacebetween',
  color: '#000',
  '& .MuiTypography-root': {
    fontWeight: 500,
  },
}));

export const PostMedia = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'column',
  width: 138,
  height: 98,
  justifyContent: 'center',
}));

export const PostRich = styled('div')({
  paddingRight: 16,
  paddingTop: 8,
  paddingBottom: 16,
  fontSize: 12,
  fontWeight: 400,
  color: '#7c7c7c',
  display: 'flex',
});

export const Video = styled('video')(({ src }) => ({
  src,
}));

export const Image = styled('img')(({ src }) => ({
  src,
}));
export const Flair = styled(RedditButton)(({ backgroundColor, flairColor }) => ({
  fontSize: 10,
  padding: '2px 10px',
  fontWeight: 'normal',
  backgroundColor,
  color: flairColor,
  '&:hover': {
    backgroundColor,
  },
  marginLeft: 10,
}));

export const PostUrlLink = styled(Box)(() => ({
  width: 135,
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
  color: 'inherit',
}));

export const PostUrl = styled('a')(({ theme }) => ({
  fontSize: 13,
  color: theme?.palette?.primary?.light,
  // height: 22,
  display: 'flex',
  alignItems: 'center',
  '&:visited ': {
    color: theme?.palette?.primary?.light,
  },
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
  // marginBottom: 10,
}));

export const LinkIcon = styled(LaunchIcon)(() => ({
  width: 13,
  height: 13,
  color: 'inherit',
}));
