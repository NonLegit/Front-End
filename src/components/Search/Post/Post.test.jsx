import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Post from './Post';

// test snapshot
test('test snapshot', async () => {
  const props = {
    post: {
      _id: 'asd212',
      author: {
        _id: 'asd212',
        userName: 'ahmed',
        profilePicture: 'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
        displayName: 'hamada',
      },
      owner: {
        icon: 'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
        _id: '12asda2',
        fixedName: 'Nonlegit',
        name: 'asdwd',
        membersCount: 123,
        description: 'this is my first subreddit',
      },
      title: 'first post',
      kind: 'self',
      text: '<p>asdas</p> <span> da </span>',
      url: 'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
      images: [
        'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
        'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
        'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
      ],
      video: 'hamada with milk',
      votes: 1200,
      commentCount: 1300,
      createdAt: '2022-11-12T12:29:28Z',
      flairId: {
        _id: 'asd',
        text: 'asd',
        backgroundColor: 'blue',
        textColor: 'red',
      },
    },
  };
  const tree = renderer.create(<Router><Post {...props} /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
