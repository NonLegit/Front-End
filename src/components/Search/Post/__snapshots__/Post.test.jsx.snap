// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-y0sa90"
>
  <div
    className="MuiBox-root css-8atqhb"
  >
    <div
      className="MuiBox-root css-11kldd9"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular css-2is7ki-MuiAvatar-root"
      >
        <img
          alt="Profile Image"
          className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2"
        />
      </div>
      <a
        className="css-kqg5s"
        color="#000"
        fontWeight="bolder"
        href="/r/Nonlegit"
        onClick={[Function]}
      >
        r/
        asdwd
      </a>
      <div
        className="MuiBox-root css-1clzuz0"
      >
        <span>
          â€¢
        </span>
        <div>
          Posted By
        </div>
        <div
          className="MuiAvatar-root MuiAvatar-circular css-2is7ki-MuiAvatar-root"
        >
          <img
            alt="Profile Image"
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2"
          />
        </div>
        <a
          className="css-1fa4cpj"
          color="inherit"
          fontWeight="normal"
          href="/user/ahmed"
          onClick={[Function]}
        >
          u/
          hamada
           
          a month ago
        </a>
      </div>
    </div>
    <a
      className="css-1h4lgge"
      href="/subreddit/Nonlegit/comments/asd212"
      onClick={[Function]}
    >
      <div
        className="MuiTypography-root MuiTypography-div css-1tw0n1b-MuiTypography-root"
      >
        first post
        <button
          backgroundColor="blue"
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-eyumte-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          flairColor="red"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          asd
        </button>
      </div>
    </a>
    <div
      className="css-1i29db6"
    >
      <span
        className="MuiTypography-root MuiTypography-span css-1askrqg-MuiTypography-root"
      >
        1.2K
         
        upvotes
         
      </span>
      <span
        className="MuiTypography-root MuiTypography-span css-1askrqg-MuiTypography-root"
      >
        1.3K
         
        comments
         
      </span>
    </div>
  </div>
</div>
`;
